src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -195,7 +195,7 @@ protected function handleCircularReference($object)
             return call_user_func($this->circularReferenceHandler, $object);
         }
 
-        throw new CircularReferenceException(sprintf('A circular reference has been detected (configured limit: %d).', $this->circularReferenceLimit));
+        throw new CircularReferenceException(sprintf('A circular reference has been detected when serializing the object of class "%s" identified by "%s" (configured limit: %d)', get_class($object), spl_object_hash($object), $this->circularReferenceLimit));
```

Comment:
It may help if we catch this exception and show spl_hash of each normalized objects to compare. Honestly I'm not convinced too. I will remove it.